### DISCUSSION TITLE
Remote host's fingerprint has changed, trying to add new host

### DISCUSSION THREAD

**Discussion by hongyucao** [open-quantum-safe/openssh-portable#141] [Q&A]
I'm having trouble setting up a connection between my server and client because my server is not listed as a known host for my client. 

I'm trying to deploy OQS-OpenSSH in a Docker container, and I've successfully built the server and client using the OQS Docker image. However, when I try to make a connection, I get an error message saying that the remote host's fingerprint has changed, and I need to add the correct host key to my known_hosts file.
`@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@    WARNING: REMOTE HOST IDENTIFICATION HAS CHANGED!     @
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
IT IS POSSIBLE THAT SOMEONE IS DOING SOMETHING NASTY!
Someone could be eavesdropping on you right now (man-in-the-middle attack)!
It is also possible that a host key has just been changed.
The fingerprint for the DILITHIUM2 key sent by the remote host is
SHA256:XcfkHncOtYtUGgKkT6S/+0n565qfLQUeGI9HxI9Ltw4.
Please contact your system administrator.
Add correct host key in /openssh/regress/known_hosts to get rid of this message.
Offending ECDSA_NISTP521_SPHINCSSHA256256FSIMPLE key in /openssh/regress/known_hosts:31
Host key for localhost-with-alias has changed and you have requested strict checking.
Host key verification failed.`

I've checked the client's known_hosts file and found that all the keys are for localhost (127.0.0.1), which won't work because this IP address is not accessible outside of the container. To solve this problem, I tried to add a new host to the known_hosts file, but I received a parse error.
`debug1: /openssh/regress/known_hosts:12: parse error in hostkeys file`
here is my attemp:
`localhost-with-alias,172.17.0.3,::1 ssh-dilithium2 XcfkHncOtYtUGgKkT6S/+0n565qfLQUeGI9HxI9Ltw4= root@c65959c4d608`

Can anyone suggest me how to format it correctly or please let me know if there is an alternative solution? I appreciate any help from you.

**Comment by xvzcf** [comment]
Perhaps you could try augmenting the `known_hosts` file with the `ssh-keyscan` utility?

**Reply by hongyucao** [reply]
Thanks for replying me!
I did try to run `ssh-keyscan 172.17.0.3 >> /openssh/regress/known_hosts`, but it resulted in `kex: algorithm: (no match)`
Is there a way to run ssh-keyscan with ssh-dilithium2?
